src/components/Input/Input.tsx

```diff
@@ -81,43 +114,66 @@ class Input extends UIComponent<any, any> {
 
   handleInputRef = c => (this.inputRef = c)
 
+  handleOnClear = e => {
+    const { clearable } = this.props
+    const { value } = this.state
+
+    if (clearable && !this.props.value && value.length !== 0) {
+      this.setState({ value: '' })
+    } else if (clearable && this.props.value && this.props.value.length !== 0) {
+      this.setState({ value: this.props.value })
+    }
```

Comment:
This can safely be replaced with:

```jsx
if (clearable) this.trySetState({ value: '' })
```

This method handles these checks and more edge cases.